src/range_proof/dealer.rs

```diff
@@ -218,6 +218,27 @@ impl<'a, 'b> DealerAwaitingProofShares<'a, 'b> {
             return Err(MPCError::WrongNumProofShares);
         }
 
+        // Validate lengths for each share
+        let mut bad_shares = Vec::<usize>::new(); // no allocations until we append
+        for (j, share) in proof_shares.iter().enumerate() {
+            match share.check_size(&self.bp_gens, j).and_then(|n| {
```

Comment:
Instead of having `check_size` return a `Result` with the actual `n` parameter, I think it would be better to have `check_size` take an `expected_n: usize` and fold the actual n == expected n check into the rest of the size check function, so that all of the consistency checking logic is in one place. This also makes the check_size calling code simpler (it might even be possible to do something like `.map_err(|_| bad_shares.push(j))` but I'm not sure if the borrow checker will be happy about a &mut closure capture).

Reply:
Good point, will do.

Reply:
<img width="728" alt="image" src="https://user-images.githubusercontent.com/698/62891440-602a6280-bd4e-11e9-9471-8274f601e471.png">